extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -311,30 +322,33 @@ public function getArticle() {
 		global $wgEnableMainPageDataMercuryApi;
 
 		try {
-			$title = $this->getTitleFromRequest();
-			$articleId = $title->getArticleId();
+			$title = $this->getRedirectTitle();
+			$data[ 'redirected' ] = true;
```

Comment:
you can create a flag - $isRedirect instead of adding to an array and then doing unset.